eng/common/scripts/modules/Package-Properties.psm1

```diff
@@ -3,22 +3,37 @@
 class PackageProps
 {
     [string]$pkgName
-    [AzureEngSemanticVersion]$pkgVersion
+    [string]$pkgVersion
     [string]$pkgDirectoryPath
     [string]$pkgServiceName
     [string]$pkgReadMePath
     [string]$pkgChangeLogPath
+    [string]$pkgGroup
 
-    PackageProps(
+    PackageProps([string]$pkgName,[string]$pkgVersion,[string]$pkgDirectoryPath,[string]$pkgServiceName)
+    {
+        $this.Initialize($pkgName, $pkgVersion, $pkgDirectoryPath, $pkgServiceName)
+    }
+
+    PackageProps([string]$pkgName,[string]$pkgVersion,[string]$pkgDirectoryPath,[string]$pkgServiceName,[string]$pkgGroup="")
+    {
+        $this.Initialize($pkgName, $pkgVersion, $pkgDirectoryPath, $pkgServiceName, $pkgGroup)
+    }
+
+    hidden [void]Initialize(
         [string]$pkgName,
         [string]$pkgVersion,
         [string]$pkgDirectoryPath,
         [string]$pkgServiceName
     )
     {
         $this.pkgName = $pkgName
-        $this.pkgVersion = [AzureEngSemanticVersion]::ParseVersionString($pkgVersion)
-        if ($this.pkgVersion -eq $null)
+        $parsedVersion = [AzureEngSemanticVersion]::ParseVersionString($pkgVersion)
```

Comment:
Why even use AzureEngSemanticVersion class at all if you are just going to pull out the RawVersion and not keep the object around. Perhaps we should let consumers of these properties parse the version if they want extra checks or incrementing ability. That would allow us to always have a version and will help with your other problem as well.

Reply:
Ok, I think I'll just leave out the use of the AzureEngSemanticVersion  class at least till it is extended for other repos?